front/src/applications/operationalStudies/hooks/useSelectedTrainAndPathForProjection.ts

```diff
@@ -0,0 +1,47 @@
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import { useEffect } from 'react';
+import { useAppDispatch } from 'store';
+
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { useSelector } from 'react-redux';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+
+const useSelectedTrainAndPathForProjection = (
```

Comment:
As the hook doesn't return anything, maybe we could have a simpler name for it ? Here are some suggestions but it can be something else

```suggestion
const useUpdateProjection = (
```

```suggestion
const useAutoUpdateProjection = (
```

Reply:
Done :white_check_mark: